File: iOS开发范例实战宝典(基础篇)/滑块/09-多个颜色的滑块/09-多个颜色的滑块/ViewController.m
//
//  ViewController.m
//  09-多个颜色的滑块
//
//  Created by coco on 16/5/31.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "ViewController.h"
#import "HJSlider.h"
@interface ViewController ()

@property (weak, nonatomic) IBOutlet HJSlider *slider;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.slider.value = 0.3;
    self.slider.midValue = 0.8;
    
    self.slider.minimumTrackTintColor = [UIColor orangeColor
                                         ];
    self.slider.maximumTrackTintColor = [UIColor blueColor];
    
    self.slider.middleTrackTintColor = [UIColor greenColor];
    
    self.slider.thumbTintColor = [UIColor purpleColor];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift练习/《iOS开发实战：图形图像、动画与多媒体卷（Swift语言版）》/第03章 图像处理/3.2__图片选择器/3.2__图片选择器/ViewController.swift
//
//  ViewController.swift
//  3.2__图片选择器
//
//  Created by coco on 16/7/4.
//  Copyright © 2016年 XHJ. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {

    @IBOutlet weak var imageV: UIImageView!
    var imagePicker : UIImagePickerController!
    
    @IBAction func photoLibrary(sender: UIButton) {
        self.changePhotoType(.SavedPhotosAlbum)
    }
    
    @IBAction func camera(sender: UIButton) {
        self.changePhotoType(UIImagePickerControllerSourceType.Camera)
    }
    
    func changePhotoType(type:UIImagePickerControllerSourceType) {
        self.imagePicker.sourceType = type
        if UIImagePickerController.isSourceTypeAvailable(type) {
            self.presentViewController(self.imagePicker, animated: true, completion: nil)
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if self.imagePicker == nil {
            self.imagePicker = UIImagePickerController()
            self.imagePicker.delegate = self
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    //delegate
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
        let image = info[UIImagePickerControllerOriginalImage]
        if image != nil {
            self.imageV.image = image! as? UIImage
        }
        picker.dismissViewControllerAnimated(true, completion: nil)
    }

    func imagePickerControllerDidCancel(picker: UIImagePickerController) {
        picker.dismissViewControllerAnimated(true, completion: nil)
    }

}

